common/roles/external-ip-type/tasks/main.yml

```diff
@@ -20,14 +20,12 @@
 
 # Backup IP address if defined
 - name: Get the backup IP address
-  when: network.backup_ip != None
   tags: facts
   set_fact:
     backup_ip: '{{ network.backup_ip }}'
 
 - name: Set backup IP address type (A or AAAA)
   tags: facts
-  when: network.backup_ip != ""
   set_fact:
     backup_ip_type: '{{ backup_ip | ipv6 | ternary("AAAA", "A") }}'
```

Comment:
I am not sure about this one. If backup_ip is None, the end result would be "A" instead of None.

Reply:
With the proposed change, if `network.backup_ip` is None (as it is by default) then `backup_ip` will be the empty string and `backup_ip_type` will be the string `"A"`, yes. It is ok since every usage of `backup_ip_type` is guarded by the condition on `backup_ip` being defined and not empty, except the following task to determine IPv6 use. Having the value "A" is ok in the following task. It works, but it is not great, I agree.

Two other options.
* Keep the proposed changes and modify the `ternary` call to get a more specific result, like:
  ```suggestion
       backup_ip_type: '{{ backup_ip | ipv6 | ternary("AAAA", backup_ip | ipv4 | ternary("A", "")) }}'
  ```
  It would mean that when `network.backup_ip` is None, then `backup_ip` and `backup_ip_type` are the empty string.
  Not very beautiful, but somewhat coherent.

* Revert the proposed changes, and change the "Set backup IP address type" conditional to check on `backup_ip`:
  ```
    when: backup_ip is defined and backup_ip != ""
  ```
  and modify the expression in the following task "Check and remember if IPv6 is used" with:
  ```
     set_fact:
      ipv6_used: >-
        (backup_ip_type is defined and backup_ip_type == "AAAA") or
        external_ip_type == "AAAA"
  ```
  That way when `network.backup_ip` is None, then `backup_ip` and `backup_ip_type` are undefined, and the `ipv6_used` fact is properly defined.
  Should work.

What do you think?